vminitd/Sources/vminitd/ManagedProcess.swift

```diff
@@ -158,6 +168,16 @@ extension ManagedProcess {
 
             // Start the underlying process.
             try command.start()
+
+            if let errorPipe = $0.errorPipe {
+                let errorReadHandle = errorPipe.fileHandleForReading
+                Task { [weak self] in
+                    if let data = try? errorReadHandle.readToEnd() {
+                        self?.state.withLock { $0.errorData = data }
+                    }
+                } 
+            }
```

Comment:
What I had in mind was we'd have a big do catch block around all of this setup, and if any of the syncPipe, ackPipe or any other setup failed, we'd try and read any error data from the error pipe. vmexec with your new setup should write any errors it encountered always, so if something went awry here we'd read it in the catch block. We want ManagedProcess.start() to throw an error if something went wrong, not store some error text in the background.